src/coprocessor/dag/expr/compare.rs

```diff
@@ -157,6 +158,20 @@ impl FnCall {
     ) -> Result<Option<Cow<'a, Json>>> {
         do_coalesce(self, |v| v.eval_json(ctx, row))
     }
+
+    pub fn like(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
+        let target = try_opt!(self.children[0].eval_string(ctx, row));
+        let target = try!(str::from_utf8(&target).map_err(Error::from));
```

Comment:
What if it's not a valid unicode string?

Reply:
Here I followed TiDB.
`children[0]` is autowrapped with `cast as string` before step in here, so I think if it's a invalid unicode string, the error has already been checked and return in `cast`?

Reply:
I don't think so. TiDB supports latin1 charset, which is not unicode compatible.

Reply:
got it. Find [something](https://stackoverflow.com/questions/28169745/what-are-the-options-to-convert-iso-8859-1-latin-1-to-a-string-utf-8/28175593#28175593) to fix this.

Reply:
Does TiDB support latin1 charset ? @shenli

Reply:
@ngaut, we only support UTF-8. Here I will rewrite a new function `eval_string_and_decode` which returns `Cow<str>`, and do string decode logic in that.